src/caretogether-pwa/src/Families/AssignmentsSection.tsx

```diff
@@ -0,0 +1,205 @@
+import { Grid, Typography, Card, CardContent, Box } from '@mui/material';
+import {
+  CombinedFamilyInfo,
+  ArrangementEntry,
+  PartneringFamilyInfo,
+} from '../GeneratedClient';
+import { format } from 'date-fns';
+import {
+  usePersonAndFamilyLookup,
+  useFamilyLookup,
+} from '../Model/DirectoryModel';
+import { partneringFamiliesData } from '../Model/ReferralsModel';
+import { useLoadable } from '../Hooks/useLoadable';
+import PersonPinCircleIcon from '@mui/icons-material/PersonPinCircle';
+import CalendarTodayIcon from '@mui/icons-material/CalendarToday';
+import { FamilyName } from '../Families/FamilyName';
+import { useNavigate } from 'react-router-dom';
+
+interface AssignmentsSectionProps {
+  family: CombinedFamilyInfo;
+}
+
+export function AssignmentsSection({ family }: AssignmentsSectionProps) {
+  const assignments = family.volunteerFamilyInfo?.assignments ?? [];
+
+  if (assignments.length === 0) return null;
+
+  return (
+    <Grid item xs={12}>
+      <Typography variant="h3" sx={{ marginBottom: 2 }}>
+        Assignments
+      </Typography>
+      <Grid container spacing={2}>
+        {assignments.map((assignment, index) => (
+          <Grid item xs={12} sm={6} md={4} key={index}>
+            <AssignmentCard assignment={assignment} />
+          </Grid>
+        ))}
+      </Grid>
+    </Grid>
+  );
+}
+
+interface AssignmentCardProps {
+  assignment: ArrangementEntry;
+}
+
+function allArrangements(partneringFamilyInfo: PartneringFamilyInfo) {
+  const results: { referralId: string; arrangement: ArrangementEntry }[] = [];
+  partneringFamilyInfo.closedReferrals?.forEach((x) =>
+    x.arrangements?.forEach((y) =>
+      results.push({ referralId: x.id!, arrangement: y })
+    )
+  );
+  partneringFamilyInfo.openReferral?.arrangements?.forEach((x) =>
+    results.push({
+      referralId: partneringFamilyInfo.openReferral!.id!,
+      arrangement: x,
+    })
+  );
+  return results;
+}
+
+function AssignmentCard({ assignment }: AssignmentCardProps) {
```

Comment:
This whole component can go in its own file, so there's no ambiguity about what's inside AssignmentsSection.tsx